internal/commands/buildpack_new.go

```diff
@@ -0,0 +1,88 @@
+package commands
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack"
+	"github.com/buildpacks/pack/internal/build"
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/logging"
+)
+
+// BuildpackNewFlags define flags provided to the BuildpackNew command
+type BuildpackNewFlags struct {
+	API     string
+	Path    string
+	Stacks  []string
+	Version string
+}
+
+// BuildpackCreator creates buildpacks
+type BuildpackCreator interface {
+	NewBuildpack(ctx context.Context, options pack.NewBuildpackOptions) error
+}
+
+// BuildpackNew generates the scaffolding of a buildpack
+func BuildpackNew(logger logging.Logger, client BuildpackCreator) *cobra.Command {
+	var flags BuildpackNewFlags
+	cmd := &cobra.Command{
+		Use:     "new <id>",
+		Short:   "Creates basic scaffolding of a buildpack.",
+		Args:    cobra.ExactValidArgs(1),
+		Example: "pack buildpack new sample/my-buildpack",
+		Long:    "buildpack new generates the basic scaffolding of a buildpack repository. It creates a new directory `name` in the current directory (or at `path`, if passed as a flag), and initializes a buildpack.toml, and two executable bash scripts, `bin/detect` and `bin/build`. ",
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			id := args[0]
+			idParts := strings.Split(id, "/")
+			dirName := idParts[len(idParts)-1]
+			_, err := os.Stat(dirName)
+			if !os.IsNotExist(err) {
+				return err
+			}
+
+			path := filepath.Join(flags.Path, dirName)
+			if len(flags.Path) == 0 {
+				cwd, err := os.Getwd()
+				if err != nil {
+					return err
+				}
+				path = filepath.Join(cwd, dirName)
+			}
+
+			var stacks []dist.Stack
+			for _, s := range flags.Stacks {
+				stacks = append(stacks, dist.Stack{
+					ID:     s,
+					Mixins: []string{},
+				})
+			}
+
+			if err := client.NewBuildpack(cmd.Context(), pack.NewBuildpackOptions{
+				API:     flags.API,
+				ID:      id,
+				Path:    path,
+				Stacks:  stacks,
+				Version: flags.Version,
+			}); err != nil {
+				return err
+			}
+
+			logger.Infof("Successfully created %s", style.Symbol(id))
+			return nil
+		}),
+	}
+
+	cmd.Flags().StringVarP(&flags.API, "api", "a", build.SupportedPlatformAPIVersions.Latest().String(), "Platform API compatibility of the generated buildpack")
+	cmd.Flags().StringVarP(&flags.Path, "path", "p", "", "Path to generate the buildpack")
+	cmd.Flags().StringVarP(&flags.Version, "version", "v", "1.0.0", "Version of the generated buildpack")
```

Comment:
I believe this line is causing a hard 💥:

```
❯ pack-dev buildpack new
panic: unable to redefine 'v' shorthand in "new" flagset: it's already used for "version" flag

goroutine 1 [running]:
github.com/spf13/pflag.(*FlagSet).AddFlag(0xc0003dc200, 0xc0003cc5a0)
	github.com/spf13/pflag@v1.0.5/flag.go:874 +0x6f0
github.com/spf13/pflag.(*FlagSet).AddFlagSet.func1(0xc0003cc5a0)
	github.com/spf13/pflag@v1.0.5/flag.go:887 +0xa8
github.com/spf13/pflag.(*FlagSet).VisitAll(0xc0000a2500, 0xc00026fc40)
	github.com/spf13/pflag@v1.0.5/flag.go:290 +0x94
github.com/spf13/pflag.(*FlagSet).AddFlagSet(0xc0003dc200, 0xc0000a2500)
	github.com/spf13/pflag@v1.0.5/flag.go:885 +0x58
github.com/spf13/cobra.(*Command).mergePersistentFlags(0xc0003da780)
	github.com/spf13/cobra@v1.1.3/command.go:1644 +0x92
github.com/spf13/cobra.(*Command).InitDefaultHelpFlag(0xc0003da780)
	github.com/spf13/cobra@v1.1.3/command.go:1018 +0x2f
github.com/spf13/cobra.(*Command).execute(0xc0003da780, 0x205cb48, 0x0, 0x0, 0xc0003da780, 0x205cb48)
	github.com/spf13/cobra@v1.1.3/command.go:776 +0x65
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003b3400, 0xc00008c000, 0xc0003b3400, 0x0)
	github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.1.3/command.go:897
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.1.3/command.go:890
main.main()
	github.com/buildpacks/pack/cmd/pack/main.go:26 +0xee

```

Reply:
Indeed. Fixed in 1874a4c (using `-V` now)